src/StreamJsonRpc/Reflection/MessageFormatterHelper.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft;
+    using Microsoft.VisualStudio.Threading;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Class containing useful methods to help on the implementation of message formatters.
+    /// </summary>
+    public class MessageFormatterHelper
```

Comment:
>MessageFormatterHelper [](start = 17, length = 22)

Let's rename this to `MessageFormatterProgressTracker` and update the xml doc comments for the type as well to indicate that this is specifically for Progress. If/when we add support for marshaling `Stream`, I'm pretty sure we're going to use a separate helper class to keep each class doing one thing.